pkg/tsdb/mssql/mssql.go

```diff
@@ -209,8 +209,8 @@ func (s *Service) CheckHealth(ctx context.Context, req *backend.CheckHealthReque
 	err = dsHandler.Ping()
 
 	if err != nil {
-		logger.Error("Check health failed", "error", err)
-		return &backend.CheckHealthResult{Status: backend.HealthStatusError, Message: err.Error()}, nil
+		transformer := mssqlQueryResultTransformer{}
```

Comment:
Rather than instantiating the transformer here, I would export `DataSourceHandler.transformQueryError` and use that one instead, like this:

https://github.com/grafana/grafana/blob/ab25ce18f46d5ad89f08899a26cb52721e2f570e/pkg/tsdb/sqleng/sql_engine.go#L110

:point_down: 

```go
func (e *DataSourceHandler) TransformQueryError(logger log.Logger, err error) error {
```

And the commented line becomes:

```go
transformer := dsHandler.TransformQueryError()
```

Because the correct query result transformer is already passed to `NewQueryDataHandler`:

https://github.com/grafana/grafana/blob/ab25ce18f46d5ad89f08899a26cb52721e2f570e/pkg/tsdb/sqleng/sql_engine.go#L124

e.g.: 

https://github.com/grafana/grafana/blob/ab25ce18f46d5ad89f08899a26cb52721e2f570e/pkg/tsdb/mssql/mssql.go#L101-L103

Also, `DataSourceHandler.transformQueryError` hides `net.Op` errors as well, which is neat.

(The same also would apply for mysql and postgres)

What do you think?

Reply:
Thanks for your review @xnyo, I addressed the changes you asked for.